client/functions.coffee

```diff
@@ -145,10 +152,17 @@
 						xhr.send form_data
 					else
 						callback and callback error,null
+			return id
 
 	delete: (path,callback) ->
 		Meteor.call "_s3_delete", path, callback
 
+	cancel: (xhrId) ->
+		req = S3.runningRequests[xhrId]
+		if req
+			req.xhr.abort()
+			S3.collection.update(req.id, {status: 'canceled'});
+
```

Comment:
This rocks! How about we put the xhr in the collection instead though? Is there an issue with that?